tools/osbuild-depsolve-dnf5

```diff
@@ -0,0 +1,500 @@
+        # between arch1 and arch2 metadata. But, as these are completely
+        # different, dnf must basically redownload everything.
+        # For downloding deltas, zchunk uses HTTP range requests. Unfortunately,
+        # if the mirror doesn't support multi range requests, then zchunk will
+        # download one small segment per a request. Because we need to update
+        # the whole metadata (10s of MB), this can be extremely slow in some cases.
+        # I think that we can come up with a better fix but let's just disable
+        # zchunk for now. As we are already downloading a lot of data when
+        # building images, I don't care if we download even more.
+        conf.zchunk = False
+
+        # Set the rest of the dnf configuration.
+        conf.module_platform_id = module_platform_id
+        conf.config_file_path = "/dev/null"
+        conf.persistdir = persistdir
+        conf.cachedir = cachedir
+
+        # Load the file lists, so dependency on paths will work
+        conf.optional_metadata_types = ['comps', 'filelists']
+
+        # Cannot modify .conf() values after this
+        self.base.setup()
+
+        try:
+            # NOTE: With libdnf5 packages are excluded in the repo setup
+            for repo in repos:
+                self._dnfrepo(repo, exclude_pkgs)
+
+            self.base.get_repo_sack().update_and_load_enabled_repos(False)
+        except RuntimeError as e:
+            raise RepoError(e)
+
+    # pylint: disable=too-many-branches
+    def _dnfrepo(self, desc, exclude_pkgs=None):
+        """Makes a dnf.repo.Repo out of a JSON repository description"""
+        if not exclude_pkgs:
+            exclude_pkgs = []
+
+        sack = self.base.get_repo_sack()
+
+        repo = sack.create_repo(desc["id"])
+        conf = repo.get_config()
+
+        if "name" in desc:
+            conf.name = desc["name"]
+
+        # At least one is required
+        if "baseurl" in desc:
+            conf.baseurl = desc["baseurl"]
+        elif "metalink" in desc:
+            conf.metalink = desc["metalink"]
+        elif "mirrorlist" in desc:
+            conf.mirrorlist = desc["mirrorlist"]
+        else:
+            raise ValueError("missing either `baseurl`, `metalink`, or `mirrorlist` in repo")
+
+        if desc.get("ignoressl", False):
+            conf.sslverify = False
+        if "sslcacert" in desc:
+            conf.sslcacert = desc["sslcacert"]
+        if "sslclientkey" in desc:
+            conf.sslclientkey = desc["sslclientkey"]
+        if "sslclientcert" in desc:
+            conf.sslclientcert = desc["sslclientcert"]
+
+        if "check_gpg" in desc:
+            conf.gpgcheck = desc["check_gpg"]
+        if "check_repogpg" in desc:
+            conf.repo_gpgcheck = desc["check_repogpg"]
+        if "gpgkey" in desc:
+            conf.gpgkey = [desc["gpgkey"]]
+        if "gpgkeys" in desc:
+            # gpgkeys can contain a full key, or it can be a URL
```

Comment:
Given that this block is non-trivial and almost identical with the existing osbuild-depsolve-dnf I wonder if we should extract a common helper here? But of course if osbuild-depsolve-dnf should not include external osbuild import then that's a nono :) Then it might be nice to cross-reference in a comment to the other implementation.

Reply:
I think it's better to just copy it. The two versions of this will never exist on the same system, and I don't think we want to make things more complex to install. It's nice to be able to just drop it into place.